find_email_addresses.py

```diff
@@ -0,0 +1,261 @@
+    """Decodes and returns a bytestring as a string
+
+    Args:
+        s: The bytestring to decode
+
+    Returns:
+        A decoded string or an empty string if encoding cannot be determined"""
+
+    enc = chardet.detect(string)
+
+    try:
+        if enc['encoding']:
+            return string.decode(enc['encoding'])
+    # Ignore pages that don't decode. These are usually low confidence anyway.
+    except UnicodeDecodeError:
+        pass
+
+    return ""
+
+def _assemble_url(link, domain, scheme):
+    """Generates a properly formatted URI
+
+    Args:
+        link: The raw link to format
+        domain: The domain to use if link does not specify
+        scheme: The scheme to use if link does not specify
+
+    Returns:
+        A properly formatted URL as a string"""
+
+    string = ''
+    url = urllib.parse.urlparse(link)
+
+    if url.scheme:
+        string += url.scheme + '://'
+    else:
+        string += scheme + '://'
+
+    if url.netloc:
+        string += url.netloc
+    else:
+        string += domain
+
+    string += url.path
+
+    if url.params:
+        string = string + ';' + url.params
+    if url.query:
+        string = string + '?' + url.query
+    if url.fragment:
+        string = string + '#' + url.fragment
+
+    return string
+
+def _is_valid_domain(domain):
+    """Determines if the provided domain is in a valid format.
+
+    Args:
+        domain: The string to test
+
+    Returns:
+        True if the domain is valid, false otherwise."""
+
+    # For some reason, urlparse won't recognize the domain without the scheme,
+    # so adding a scheme is a bit of a hack, but it works. It will even work
+    # if someone enters "http://domain.com" as the domain.
+    url = urllib.parse.urlparse("http://" + domain)
+    if url.path or url.params or url.query or url.fragment or not url.netloc:
+        return False
+
+    return True
+
+def get_emails_in_domain(domain, scheme='http', verbosity=0):
+    """Returns email addresses found in domain.
+
+    Iterates through publicly accessible pages/files found at
+    <scheme>://<domain> and collects all email addresses.
+
+    Args:
+        domain: The domain to search as a string
+        scheme: The scheme to use (default: http) as a string
+        verbosity: The level of verbosity (default: 0, max: 2)
+
+    Returns:
+        A list containing unique email addresses found
+
+    Raises:
+        DomainError: 'domain' is not a valid domain."""
+
+    if not _is_valid_domain(domain):
+        raise DomainError('"{}" is not a valid domain.'.format(domain))
+
+    emails = deque()
+    pages_visited = deque()
+    pages_to_visit = deque(['/'])
+
+    while len(pages_to_visit):
+        page_contents = ''
+
+        path = pages_to_visit.popleft()
+        link = _assemble_url(path, domain, scheme)
+        try:
+            if verbosity >= 2:
+                print('Processing page "{}"...'.format(link))
+            with urllib.request.urlopen(link) as page:
```

Comment:
If I run this script on `mcent.com` (this is the landing page site for our app) it finds the following emails
{{{
apps@freecharge.com
support@tapcash.com
care@paytm.com
info@easymobilerecharge.com
help@getrewards.mobi
}}
which do not appear on our site.  This because this link `http://mcent.com/app?tc=PJYFLL` results in a 301 redirect to an external site (the google play store).  Can you update this part of the code to ignore redirects off the target domain?